src/providers/deno.rs

```diff
@@ -0,0 +1,37 @@
+use super::Provider;
+use crate::{nixpacks::app::App, providers::Pkg};
+use anyhow::Result;
+use regex::Regex;
+
+pub struct DenoProvider {}
+
+impl Provider for DenoProvider {
+    fn name(&self) -> &str {
+        "deno"
+    }
+
+    fn detect(&self, app: &App) -> Result<bool> {
+        if !app.includes_file("src/index.ts") {
+            false;
+        }
+
+        let re = Regex::new(r##"(?m)^import .+ from "https://deno.land/[^"]+\.ts";?$"##).unwrap();
+        Ok(app.find_match(&re, "**/*.ts"))
+    }
+
+    fn pkgs(&self, _app: &App) -> Vec<Pkg> {
+        vec![Pkg::new("pkgs.stdenv"), Pkg::new("pkgs.deno")]
+    }
+
+    fn install_cmd(&self, _app: &App) -> Result<Option<String>> {
+        Ok(None)
+    }
+
+    fn suggested_build_cmd(&self, _app: &App) -> Result<Option<String>> {
+        Ok(None)
+    }
+
+    fn suggested_start_command(&self, _app: &App) -> Result<Option<String>> {
+        Ok(Some("deno run src/index.ts".to_string()))
```

Comment:
Is this going to handle the explicit flags Deno needs, like `--allow-net`?

Reply:
Yeah good point. I think it should allow-net by default. We can eventually use environment variables to control addition Deno behaviour.

Reply:
It does not. I wonder if we should run with `-A` like this [Heroku Deno builpack](https://github.com/chibat/heroku-buildpack-deno/blob/master/bin/compile) does since we don't know what permissions the user will actually want.

Reply:
I mean, granting -A is generally dangerous, but as I don't see a better way of doing this, that's probably what is going to have to be done.